__tests__/timeoutLink.test.ts

```diff
@@ -1,11 +1,39 @@
-import TimeoutLink from '../src/timeoutLink';
-import { ApolloLink, execute, Observable, HttpLink } from '@apollo/client/core';
+import {
+  ApolloClient,
+  ApolloLink,
+  type GraphQLRequest,
+  HttpLink,
+  InMemoryCache,
+  Observable,
+  execute,
+} from '@apollo/client/core';
 import gql from 'graphql-tag';
+import TimeoutLink from '../src/timeoutLink';
+import type { FetchResult } from '../src/types';
+
 
 const TEST_TIMEOUT = 100;
 
 const timeoutLink: TimeoutLink = new TimeoutLink(TEST_TIMEOUT);
 
+let apolloClientInstance: ApolloClient | null = null;
+const executeWrapper = (link: ApolloLink, operation: GraphQLRequest) => {
+  if (execute.length === 3 && !apolloClientInstance) {
+    const httpLink = new HttpLink({ uri: "/graphql" });
+    const timeoutHttpLink = timeoutLink.concat(httpLink);
+    apolloClientInstance = new ApolloClient({
+      link: timeoutHttpLink,
+      cache: new InMemoryCache(),
+    });
+  }
+  return execute.length === 3
+    // @ts-ignore
+    ? execute(link, operation, { client: apolloClientInstance })
```

Comment:
execute in `@apollo/client@4` requires passing the client that initiating the execute, so this wrapper acts as a shim around that to support both 3 and 4.